### PR TITLE
chore: sidebar refactoring & permissions added

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/HyperSwitch/UserManagement/PermissionHelper.res [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+let getAccessValue = (~permissionValue: permissionType, permissionList) => {
+  open AuthTypes
+  let isPermissionFound =
+    permissionList
+    ->Array.find(ele => {
+      ele === permissionValue
+    })
+    ->Option.getWithDefault(UnknownPermission(""))
+
+  isPermissionFound->mapPermissionTypeToString->String.length > 0 ? Read : NoAccess
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/UserManagement/PermissionHelper.res:109]: Can we use switch here than checking the string length ?

Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/UserManagement/PermissionHelper.res:109]: Updated

File: src/screens/HyperSwitch/UserManagement/PermissionHelper.res [hunk 2]
```diff
@@ -0,0 +1,110 @@
+type permissionType =
```
Comment (jainlokesh318) [inline src/screens/HyperSwitch/UserManagement/PermissionHelper.res:1]: naming standards

Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/UserManagement/PermissionHelper.res]: Resolved

**Review by jainlokesh318 — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**